dev-libs/libbase58/libbase58-0.1.4.ebuild

```diff
@@ -11,18 +11,37 @@ LICENSE="MIT"
 SRC_URI="https://github.com/luke-jr/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 SLOT="0/0"
 KEYWORDS="amd64 ~arm ~mips ~ppc ~ppc64 x86"
-IUSE="tools"
+IUSE="test tools"
+RESTRICT="!test? ( test )"
 
-DEPEND=""
-RDEPEND="${DEPEND}"
+# NOTE: If not testing, we don't need non-native libgcrypt
+RDEPEND="tools? ( dev-libs/libgcrypt )"
+DEPEND="${RDEPEND}
+	test? ( dev-libs/libgcrypt[${MULTILIB_USEDEP}] )
+"
+
+ECONF_SOURCE="${S}"
 
 src_prepare() {
 	eapply_user
 	eautoreconf
 }
 
-src_configure() {
-	econf \
-		--disable-static \
-		$(use_enable tools tool)
+multilib_src_configure() {
+	local myeconf=(
+		--disable-static
+		LIBGCRYPT_CONFIG="${EPREFIX}/usr/bin/${CHOST}-libgcrypt-config"
+	)
+	if multilib_is_native_abi && use tools; then
+		myeconf+=( --enable-tool )
+	elif use test; then
+		myeconf+=( --enable-tool --bindir='/TRASH' )
+	else
+		myeconf+=( --disable-tool )
+	fi
```

Comment:
I'd maybe find it easier to just write `REQUIRED_USE="test? ( tools )`
But make sure you fix the test failure with appending temporary PATH so it finds the compiled binary before it's installed in root.

Reply:
`tools` only requires libgcrypt natively built, whereas `test` requires it for all target ABIs.

I'm going to have to fix the test issue upstream somehow.

Reply:
I get it. Could you clean this ebuild and `RESTRICT="test"` on this version so we get this PR moved forward, while you fix it upstream?

Reply:
Building in the source dir works, so simply using `multilib_copy_sources` for this version seems fine.